### PR TITLE
[1LP][RFR] automated test_vpc_env_selection

### PR DESCRIPTION
**Pull request by lina-nikiforova**
{{ pytest: -v cfme/tests/cloud/test_providers.py::test_vpc_env_selection  --use-provider=complete  }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 2e6c235244d3199867f69358e507f5ebcb81ee22:

`cfme/tests/cloud/test_providers.py`:
- :red_circle: Line 12:1: **F401** *'widgetastic.utils.partial_match' imported but unused*
- :bangbang: Line 868:63: **E226** *missing whitespace around arithmetic operator*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/cloud/test_providers.py [hunk 1]
```diff
@@ -827,6 +827,48 @@ def test_domain_id_validation(request, provider):
...
+    Test selection of components in environment page of cloud instances
+    with selected virtual private cloud
+
+    Polarion:
+        assignee: anikifor
+        casecomponent: WebUI
+        initialEstimate: 1/2h
+        testSteps:
+            1. Provision an Azure Instance from an Image.
+            2. At the environment page, try to select components without vpc with vpc
```
Comment (sshveta) [inline cfme/tests/cloud/test_providers.py]: This should be : 
2. At the environment page, try to select components with vpc and without vpc

File: cfme/tests/cloud/test_providers.py [hunk 2]
```diff
@@ -827,6 +827,48 @@ def test_domain_id_validation(request, provider):
...
+    Bugzilla:
+        1315945
+    """
+    vm_name = random_vm_name('prov-az')
+    template = provisioning.get('image').get('name')
+    vm = appliance.collections.cloud_instances.instantiate(name=vm_name,
+                                                           provider=provider,
+                                                           template_name=template)
+    # default args select vpc
+    data = vm.vm_default_args
```
Comment (sshveta) [inline cfme/tests/cloud/test_providers.py]: Question : default args what values for vpc ?
This is not covering test without vpc .

File: cfme/tests/cloud/test_providers.py [hunk 3]
```diff
@@ -827,6 +827,48 @@ def test_domain_id_validation(request, provider):
     assert view.flash.messages[0].type == 'error'
 
 
+@test_requirements.azure
+@pytest.mark.provider([AzureProvider], override=True, selector=ONE)
```
Comment (sshveta) [inline cfme/tests/cloud/test_providers.py]: Required : @pytest.mark.meta(automates=[BZ(1315945)])

File: cfme/tests/cloud/test_providers.py (hunk 1 above)
Comment (lina-nikiforova) [reply cfme/tests/cloud/test_providers.py]: replaced with `2. At the environment page, try to select components with vpc`.

this test should cover only provisioning with vpc, some other provisioning test provisions instances without selecting vpc

File: cfme/tests/cloud/test_providers.py (hunk 2 above)
Comment (lina-nikiforova) [reply cfme/tests/cloud/test_providers.py]: There's `cloud_network` and `cloud_subnet` args that we need in this test.
This test needs to cover only "with vpc" as the "without" is covered by provisioning test.


image

**Comment by lina-nikiforova** [comment]
@sshveta please take a look, I addressed your comments.

**Pull request merged**